### DISCUSSION TITLE
Add Spike Based Loss Methods for Multi Label Classification?

### DISCUSSION THREAD

**Discussion by Michaeljurado42** [lava-nc/lava-dl#219] [Ideas]
Hello and good day all!

My team and I recently released a paper in which we introduced a spike-based extension to sigmoid, mentioned [here](https://ieeexplore.ieee.org/abstract/document/10191787). The idea is very similar to [Spikemax](https://ieeexplore.ieee.org/abstract/document/9892379/) released last year. Here is a graphic of this new function:

<img src="https://github.com/lava-nc/lava-dl/assets/18356610/356f409b-b559-43ea-a7c3-16b7752f482e" width="500" />

As mentioned in the title, this function is useful when you need to perform multi-label classification with multiple classes available in a single data sample. To test our function, we [overlayed MNIST imagery](https://youtu.be/2AVVoHsKWuU) to see if the SNN could successfully perform multi-label classification.

Another difference between this function and Spikemax involves the use of a scaling parameter before the non-linearity. In the paper, we also found that training progressed much more quickly with the same experimental setup when we added this parameter to Spikemax.

Anyhow, please let me know if this would be a feature that you would be interested in incorporating into the codebase. I can create a pull request based on our [code](https://github.com/audreydunn/lava-dl/blob/spikemoid_public/src/lava/lib/dl/slayer/loss.py#L235) if so.

**Comment by bamsumit** [comment]
@Michaeljurado42, yes please, we would be happy to have it included. :)

**Reply by Michaeljurado42** [reply]
Oh thank you for the quick reply. I'll get on it then!

**Reply by Michaeljurado42** [reply]
@bamsumit Alright its up. Although I think the modifications to Spikemax are also worth including at some point. I just stuck with adding Spikemoid for now.
